packages/amplication-client/src/Resource/git/AuthResourceWithBitbucketCallback.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from "react";
+import { gql, useMutation } from "@apollo/client";
+import { useTracking } from "../../util/analytics";
+import { AnalyticsEventNames } from "../../util/analytics-events.types";
+import { EnumGitProvider } from "../../models";
+
+const AuthResourceWithBitbucketCallback = () => {
+  const { trackEvent } = useTracking();
+  const [completeAuthWithGit] = useMutation<boolean>(COMPLETE_OAUTH2_FLOW, {
+    onCompleted: (data) => {
+      window.opener.postMessage({ completed: true });
+      // close the popup
+      window.close();
+    },
+  });
+
+  useEffect(() => {
+    const queryString = window.location.search;
```

Comment:
to my opinion the first three line are useEffect by itself, then the rest is another one

Reply:
because your useEffect doesnt listen to url param

Reply:
it is listening to the code query param.